gradle/gradle-mvn-push.gradle

```diff
@@ -81,6 +81,7 @@ afterEvaluate { project ->
     if (plugins.hasPlugin('com.android.application') || plugins.hasPlugin('com.android.library')) {
       task androidJavadocs(type: Javadoc) {
         source = android.sourceSets.main.java.srcDirs
+        exclude "**/internal/**"
```

Comment:
The javadoc talk failed because dependencies other than Android aren't in the javadoc classpath. This doesn't fix that, but only internal classes used code from external dependencies, and we shouldn't generate javadoc for internal classes anyway. We might still need to actually fix the javadoc task in the future though.

Reply:
You can add them back. https://github.com/vanniktech/gradle-maven-publish-plugin/blob/0b2c99e74f00da096a2a8a7ff27b51ead5009ac2/src/main/groovy/com/vanniktech/maven/publish/MavenPublishPlugin.groovy#L102

```
// Append also the classpath and files for release library variants. This fixes the javadoc warnings.
// Got it from here - https://github.com/novoda/bintray-release/pull/39/files
def releaseVariant = project.android.libraryVariants.toList().last()
classpath += releaseVariant.javaCompile.classpath
classpath += releaseVariant.javaCompile.outputs.files
```

Reply:
nice!

Reply:
🙏https://github.com/square/leakcanary/pull/1012

Reply:
Added comment relating to this snippet here: https://github.com/square/leakcanary/pull/1012#discussion_r196144602